### PR TITLE
Extract response utils

### LINKED ISSUE #186: Extract response function into a util
We have some methods in the authenticator package that handle HTTP
responses. We can benefit from extracting them into utils with code
clarity.

DoD:
- [ ] The methods `readBody` & `validateResponse` are extracted into a util
- [ ] These methods have UTs

### PR DESCRIPTION
**Pull request by orenbm**
### What does this PR do?
We have some methods in the authenticator package that handle HTTP
responses. We can benefit from extracting them into utils with code
clarity.

### What ticket does this PR close?
Resolves #186

#### Change log
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes

#### Documentation
- [x] This PR does not require updating any documentation

### CODE REVIEW

**Review by abrahamko — APPROVED** [review]
Good separation!
:+1:

**Comment by orenbm** [comment]
Created [an issue](https://github.com/cyberark/conjur-authn-k8s-client/issues/188) for adding UTs

File: pkg/utils/response.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package utils
+
+import (
+	"io/ioutil"
+	"net/http"
+
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/log"
+)
+
+// ReadResponseBody returns the response body
```
Comment (eranha) [inline pkg/utils/response.go:10]: Seems redundant.

File: pkg/utils/response.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+		return nil, log.RecordedError(log.CAKC022, err)
+	}
+
+	return responseBytes, err
+}
+
+// validateResponse checks the HTTP status of the response. If it's less than
+// 300, it returns the response body as a byte array. Otherwise it returns
+// a NewResponseError.
+func ValidateResponse(resp *http.Response) error {
```
Comment (eranha) [inline pkg/utils/response.go:25]: so the name should be `ValidateResponseCode`

**Review by eranha — CHANGES_REQUESTED** [review]

**Pull request merged**